packages/subiquity_client/lib/subiquity_server.dart

```diff
@@ -53,17 +53,23 @@ abstract class SubiquityServer {
 
   Future<String> start(ServerMode serverMode, [List<String>? args]) async {
     final socketPath = _getSocketPath(serverMode);
-    if (!_shouldStart(serverMode)) {
-      return socketPath;
+    if (_shouldStart(serverMode)) {
+      var subiquityCmd = <String>[
+        '-m',
+        _pythonModule,
+        if (serverMode == ServerMode.DRY_RUN) '--dry-run',
+        ...?args,
+      ];
+      _startSubiquity(subiquityCmd);
```

Comment:
Shouldn't we be `await`ing the result of `_startUbiquity(…)` here? Otherwise `_waitSubiquity(…)` risks starting to poll the server too early, no?

Reply:
Yes, it's definitely the right thing to do. Thanks!